x/mongo/driver/topology/pool.go

```diff
@@ -414,6 +412,12 @@ func (p *pool) closeConnection(c *connection) error {
 	delete(p.opened, c.poolID)
 	p.Unlock()
 
+	if atomic.LoadInt32(&c.connected) == connected {
+		c.waitForContext()
+		c.cancelConnect()
```

Comment:
can collapse these two into an actual cancelConnect function that waits and then cancels

Reply:
done